Dockerfile

```
@@ -47,19 +53,29 @@ ENV CARGO_UNSTABLE_BUILD_STD="std,panic_abort"
 # chef stage's copy at /; the nightly-only RUSTFLAGS above depends on it.
 COPY rust-toolchain.toml .
 COPY --from=planner /app/recipe.json recipe.json
-# build-std demands an explicit --target; use the image's own host triple so
-# multi-arch builds (e.g. buildx linux/arm64) keep producing native binaries
-# exactly like the implicit-target build did.
+# build-std demands an explicit --target; derive the image's own host triple so
+# buildx per-arch builds (linux/amd64, linux/arm64) each target their own arch.
 RUN rustc -vV | sed -n 's/^host: //p' > /rust-target && test -s /rust-target
```

Hadolint (2.14.0)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)